OpenRA.Mods.Common/Traits/HitShape.cs

```diff
@@ -0,0 +1,60 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.HitShapes;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Shape of actor for damage calculations.")]
+	public class HitShapeInfo : ConditionalTraitInfo
+	{
+		[FieldLoader.LoadUsing("LoadShape")]
+		public readonly IHitShape Shape;
+
+		static object LoadShape(MiniYaml yaml)
+		{
+			IHitShape ret;
+
+			var shapeNode = yaml.Nodes.Find(n => n.Key == "Shape");
+			var shape = shapeNode != null ? shapeNode.Value.Value : string.Empty;
+
+			if (!string.IsNullOrEmpty(shape))
+			{
+				ret = Game.CreateObject<IHitShape>(shape + "Shape");
```

Comment:
This probably wants to live inside the try so we can give a sensible exception if the shape type doesn't exist.